search/lib/api/provider.js

```diff
@@ -71,12 +150,45 @@ async function getProviders (request, response) {
   response.status(200).json(providerCatalog);
 }
 
+/**
+ * Get all the providers.
+ *   Note: We can't query cloud holding providers, so this will return all the providers.
+ *         But it will return the links to cloud holding collections for these providers.
+ *         It is possible that a link returns nothing if that provider doesn't contain cloud holding collections.
+ */
+async function getCloudProviders (request, response) {
+  const event = request.apiGateway.event;
+  const providers = await cmr.getProviderList();
```

Comment:
Do we have access to a cache value for this? Is that function cached?

Reply:
I just checked. it's not cached.

Reply:
I don't think the  `getProviders` and `getCloudProviders` functions benefit much from caching. They are only called when one visits the root endpoint, as it needs to list all the providers which is just a single call to CMR (/ingest/providers.json) and not a terribly big response.

I'll point out though, as mentioned elsewhere, these two functions are nearly identical if not for the URLs to the individual providers which are formed as `/cloudstac` rather than `/stac`.  I think if a way could be figured out for these two functions to be consolidated much of the other duplicated functions for cloudstac will fall into place.

Reply:
getCloudProviders and getProviders are the easiest to be combined because they are the shortest and only the link and title, description are different. Other functions are not as easy and I don't think it has much to do with these two functions combined or not.

Reply:
Taking getCloudProvider and getProvider as an example, the getCloudProvider calls cmr.findCollections and getProvider calls cmr.getProvider; getCloudProvider needs to go through pagination, getProvider doesn't.  Also the links are different.